compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/TypeDefBuilderHelper.java

```diff
@@ -319,6 +302,7 @@ public static void populateStructureFields(Types types, SymbolTable symTable,
                                                long flag, boolean isImmutable) {
         BTypeSymbol structureSymbol = structureType.tsymbol;
         LinkedHashMap<String, BField> fields = new LinkedHashMap<>();
+        structureType.typeInclusions = origStructureType.typeInclusions;
```

Comment:
Why did we need to add this? May have to rename the method too, since it does more than populate structure fields.

Reply:
```ballerina
type Foo record {|
    int[] x = [1];
    int[] y = [];
|};

type Bar record {|
    *Foo;
    int[] & readonly x = [2];
|};

public function  main() {
    Bar & readonly b1 = {};
    io:println(b1);
}
```

When considering the above scenario , we need to create an immutable type for 'Bar.' When doing that, we need to include type inclusions into the newly generated type. For that I do the above change.

Reply:
Change function name as `populateStructureFieldsAndTypeInclusions`